environment/vscode-ssh.rst

```diff
@@ -0,0 +1,410 @@
+
+Type the password associated with your CNetID and press enter (nothing will appear on the screen as you type your password, but this is normal; your keypresses are still being registered).
+
+You should see a message about when you last logged on, followed by a prompt that looks like
+
+::
+
+    username@linuxX:~$
+
+where ``username`` is replaced by your CNetID, and `X` is replaced by a number from 1 to 7. You are now connected to the Linux computers on campus. Try running a few terminal commands, like ``pwd``, ``ls`` and ``cd``. If you already did the `Virtual Linux lab`_, you should be able to find the files that you created for it.
+
+Type ``logout`` and press enter to close your connection to the campus Linux computers. Type ``exit`` again and press enter to exit Windows PowerShell.
+
+macOS
+~~~~~
+
+An SSH client comes pre-installed. However, you should check that it works as expected before moving on.
+
+Press Command-Space to open Spotlight Search. Begin typing *Terminal*, and click on the option when it appears.
+
+.. figure:: code-img/install-ssh-mac-1.png
+
+At the terminal prompt, type
+
+::
+
+    ssh username@LINUX_SERVER
+
+where ``username`` should be replaced by your CNetID and
+``LINUX_SERVER`` should be replaced with your assigned linux server.
+
+.. figure:: code-img/install-ssh-mac-2.png
+
+You should be prompted for your password. Type the password associated with your CNetID and press enter (nothing will appear on the screen as you type your password, but this is normal; your keypresses are still being registered).
+
+You should see a message about when you last logged on, followed by a prompt that looks like
+
+::
+
+    username@linuxX:~$
+
+where ``username`` is replaced by your CNetID, and `X` is replaced by a number from 1 to 7. You are now connected to the Linux computers on campus. Try running a few terminal commands, like ``pwd``, ``ls`` and ``cd``.
+
+Type ``logout`` and press enter to close your connection to the campus Linux computers and return to your own computer's terminal prompt.
+
+
+Linux
+~~~~~
+
+.. These instructions are essentially copied from https://code.visualstudio.com/docs/remote/troubleshooting#_installing-a-supported-ssh-client .
+
+Debian/Ubuntu: Run ``sudo apt-get install openssh-client``
+
+RHEL/Fedora/CentOS: Run ``sudo yum install openssh-clients``
+
+After installing, you should verify that you can connect to the Linux computers on campus. In the terminal, type,
+
+::
+
+    ssh username@LINUX_SERVER
+
+where ``username`` is replaced by your CNetID and ``LINUX_SERVER`` is replaced with your assigned linux server. You should be prompted for the password associated with your CNetID. Then you should be able to run terminal commands on the campus Linux computers.
+
+Step 3: Install Extensions for VSCode
```

Comment:
If this is meant to be a general use guide, perhaps we should remove this section on C/C++ extensions. Or perhaps move it to the end of the article as a general guide on adding extensions for all languages.

Reply:
Fixed in branch `vaa/tutorial` (I generalized it so it doesn't refer just to C/C++)